horde/apis/v2/base.py

```diff
@@ -1774,7 +1774,7 @@ def patch(self, sharedkey_id=''):
             raise e.InvalidAPIKey("patch sharedkey")
         if sharedkey.user_id != user.id:
             raise e.Forbidden(f"Shared Key {sharedkey.id} belongs to {sharedkey.user.get_unique_alias()} and not to {user.get_unique_alias()}.")
-        if not self.args.expiry and not self.args.kudos and not self.arg.name:
+        if not self.args.expiry and not self.args.kudos and not self.args.name:

```

Comment:
This check will mean any attempt to modify `max_image_pixels`, `max_image_steps`, and `max_pixel_tokens` will be missed further down the conditional chain, unless you send kudos/expiry/name in addition to the other values.

Inelegant validation solution equivalent fix:
```suggestion
        if None in [self.args.expiry, self.args.kudos, self.args.name, self.args.max_image_pixels, self.args.max_image_steps, self.args.max_text_tokens]:
```

As a general Python tip that catches everyone sometime... `not x` will match integer 0 and empty string as false, if you want to check for specifically unset values you need to check for `if x is None` or `if x is not None`.

```
not 0 = True
not 1 = False
not True = False
not False = True
not "" = True
not "string" = False
```

Reply:
Oops my code there is wrong, one sec...